code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java

```diff
@@ -92,6 +95,34 @@ public boolean isError() {
         return false;
     }
 
+    public boolean isHostActive() {
+        for (DeploymentUnitInstance deployUnitInstance : getDeploymentUnitInstances()) {
+            if (deployUnitInstance instanceof ExternalDeploymentUnitInstance) {
```

Comment:
I felt the common case should have an instance and ExternalDeploymentUnitInstance is more the exception.  And that's why I had it throw a RuntimeException in its implementation.